frame/collective/src/lib.rs

```diff
@@ -91,195 +97,64 @@ pub trait DefaultVote {
-		/// The prime member that helps determine the default vote behavior in case of absentations.
-		pub Prime get(fn prime): Option<T::AccountId>;
+		/// Weight information for extrinsics in this pallet.
+		type WeightInfo: WeightInfo;
 	}
-	add_extra_genesis {
-		config(phantom): sp_std::marker::PhantomData<I>;
-		config(members): Vec<T::AccountId>;
-		build(|config| Module::<T, I>::initialize_members(&config.members))
-	}
-}
 
-decl_event! {
-	pub enum Event<T, I=DefaultInstance> where
-		<T as frame_system::Config>::Hash,
-		<T as frame_system::Config>::AccountId,
-	{
-		/// A motion (given hash) has been proposed (by given account) with a threshold (given
-		/// `MemberCount`).
-		/// \[account, proposal_index, proposal_hash, threshold\]
-		Proposed(AccountId, ProposalIndex, Hash, MemberCount),
-		/// A motion (given hash) has been voted on by given account, leaving
-		/// a tally (yes votes and no votes given respectively as `MemberCount`).
-		/// \[account, proposal_hash, voted, yes, no\]
-		Voted(AccountId, Hash, bool, MemberCount, MemberCount),
-		/// A motion was approved by the required threshold.
-		/// \[proposal_hash\]
-		Approved(Hash),
-		/// A motion was not approved by the required threshold.
-		/// \[proposal_hash\]
-		Disapproved(Hash),
-		/// A motion was executed; result will be `Ok` if it returned without error.
-		/// \[proposal_hash, result\]
-		Executed(Hash, DispatchResult),
-		/// A single member did some action; result will be `Ok` if it returned without error.
-		/// \[proposal_hash, result\]
-		MemberExecuted(Hash, DispatchResult),
-		/// A proposal was closed because its threshold was reached or after its duration was up.
-		/// \[proposal_hash, yes, no\]
-		Closed(Hash, MemberCount, MemberCount),
-	}
-}
+	#[pallet::pallet]
+	#[pallet::generate_store(pub(super) trait Store)]
+	pub struct Pallet<T, I=()>(PhantomData<(T, I)>);
 
-decl_error! {
-	pub enum Error for Module<T: Config<I>, I: Instance> {
-		/// Account is not a member
-		NotMember,
-		/// Duplicate proposals not allowed
-		DuplicateProposal,
-		/// Proposal must exist
-		ProposalMissing,
-		/// Mismatched index
-		WrongIndex,
-		/// Duplicate vote ignored
-		DuplicateVote,
-		/// Members are already initialized!
-		AlreadyInitialized,
-		/// The close call was made too early, before the end of the voting.
-		TooEarly,
-		/// There can only be a maximum of `MaxProposals` active proposals.
-		TooManyProposals,
-		/// The given weight bound for the proposal was too low.
-		WrongProposalWeight,
-		/// The given length bound for the proposal was too low.
-		WrongProposalLength,
-	}
-}
+	/// Origin for the collective module.
+	#[pallet::origin]
+	pub type Origin<T, I = ()> = RawOrigin<<T as frame_system::Config>::AccountId, I>;
 
-/// Return the weight of a dispatch call result as an `Option`.
-///
-/// Will return the weight regardless of what the state of the result is.
-fn get_result_weight(result: DispatchResultWithPostInfo) -> Option<Weight> {
```

Comment:
why did you move this method to inside the Pallet ?
I think better not to do these refactor in this PR